ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,209 @@
+    run_cmd(login_cmd)
+    logger.info(f"podman login success for kubeadmin user!!")
+
+
+def validate_pvc_mount_on_registry_pod():
+    """
+    Function to validate pvc mounted on the registry pod
+    """
+    pod_obj = get_registry_pod_obj()
+    mount_point = pod_obj[0].exec_cmd_on_pod(command="mount")
+    assert "/registry" in mount_point, f"pvc is not mounted on pod {pod_obj.name}"
+    logger.info("Verified pvc is mounted on image-registry pod")
+
+
+def validate_registry_pod_status():
+    """
+    Function to validate registry pod status
+    """
+    pod_obj = get_registry_pod_obj()
+    helpers.wait_for_resource_state(pod_obj[0], state=constants.STATUS_RUNNING)
+
+
+def validate_all_image_registry_pod_status():
+    """
+    Function to validate all image-registry pod status
+    """
+    pass
+
+
+def validate_data_at_backend():
+    """
+    Function to validate image_push data from ceph
+    """
+    pass
+
+
+def get_registry_pvc():
+    """
+    Function to get registry pvc
+    """
+    pod_obj = get_registry_pod_obj()
+    return pod.get_pvc_name(pod_obj)
+
+
+def get_default_route_name():
+    """
+    Function to get default route name
+    Return:
+        route_name (str): Returns default route name
+    """
+    ocp_obj = ocp.OCP()
+    route_cmd = f"get route -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE} -o yaml"
+    route_dict = ocp_obj.exec_oc_cmd(command=route_cmd)
+    return route_dict.get('items')[0].get('spec').get('host')
+
+
+def add_role_to_user(role_type, user):
+    """
+    Function to add role to user
+    Args:
+        role_type (str): Type of the role to be added
+        user (str): User to be added for the role
+    """
+    ocp_obj = ocp.OCP()
+    role_cmd = f"policy add-role-to-user {role_type} {user} -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE}"
+    assert ocp_obj.exec_oc_cmd(command=role_cmd), f"Adding role failed"
+    logger.info(f"Role_type {role_type} added to the user {user}")
+
+
+def enable_route_and_create_ca_for_registry_access():
+    """
+    Function to enable route and to create ca and copy to respective location for registry access
+    """
+    ocp_obj = ocp.OCP(
+        kind=constants.CONFIG, namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+    )
+    assert ocp_obj.patch(
+        resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME,
+        params='{"spec": {"defaultRoute": true}}', type='merge'
+    ), f"Registry pod defaultRoute enable is not success"
+    logger.info(f"Enabled defaultRoute to true")
+    ocp_obj = ocp.OCP()
+    crt_cmd = f"get secret {constants.DEFAULT_ROUTE_CRT} -n {constants.OPENSHIFT_INGRESS_NAMESPACE} -o yaml"
+    crt_dict = ocp_obj.exec_oc_cmd(command=crt_cmd)
+    crt = crt_dict.get('data').get('tls.crt')
+    route = get_default_route_name()
+    with open(f"/tmp/{route}.crt", "wb") as temp:
+        temp.write(base64.b64decode(crt))
+    run_cmd(f"sudo cp /tmp/{route}.crt /etc/pki/ca-trust/source/anchors/{route}.crt")
```

Comment:
I guess no one wants to have this running on their systems! No sudo commands should be running on the machine where ocs-ci is running and it will fail on the systems when running locally as it will ask for password. In jenkins slaves we don't ask for password so there it will probably work but I think we don't want this to happens there as well.   Can you explain why it's needed and if it can be achieved somehow differently?

Is the route certificate dynamic for each run or is this some specific cert which is same? If second this step should be mentioned as prerequisite and user should do this action by himself. On jenkins we can add it to our playbook with other certs we adding.  (We are using this role: https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/tree/master/ansible/roles/copy-redhat-certs)

Reply:
ack, There is another way to perform this, without using sudo let me explore and update the code.

Reply:
not using sudo any more, resolving this comment.